lab_3/main.py

```diff
@@ -29,6 +81,8 @@ class LetterStorage:
     Stores and manages letters
     """
 
+    free_id = 1
```

Comment:
This would be a *class* field that would be shared across all instances. I suggest putting it in the constructor so it would explicitly belong to the current instance.